<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Application de Traitement d'Images</string>
  </property>
  <widget class="QWidget" name="centralwidget">

   <!-- Bouton pour charger une image -->
   <widget class="QPushButton" name="ChargerImage">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>120</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Charger Image</string>
    </property>
   </widget>

   <!-- Vue pour afficher l'image -->
   <widget class="QGraphicsView" name="AfficherImage">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>10</y>
      <width>300</width>
      <height>250</height>
     </rect>
    </property>
   </widget>

   <!-- Vue pour afficher les résultats -->
   <widget class="QGraphicsView" name="AffichageResultat">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>10</y>
      <width>300</width>
      <height>250</height>
     </rect>
    </property>
   </widget>

   <!-- Boutons pour calculer l'histogramme -->
   <widget class="QPushButton" name="CalculerHisto">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>120</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Calculer Histogramme</string>
    </property>
   </widget>

   <widget class="QCheckBox" name="Canal_R">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>100</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Canal Rouge</string>
    </property>
   </widget>

   <widget class="QCheckBox" name="Canal_V">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>130</y>
      <width>100</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Canal Vert</string>
    </property>
   </widget>

   <widget class="QCheckBox" name="Canal_B">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>160</y>
      <width>100</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Canal Bleu</string>
    </property>
   </widget>

   <!-- Boutons pour détection des contours -->
   <widget class="QPushButton" name="DetectionContour">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>120</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Détection Contours</string>
    </property>
   </widget>

   <!-- Boutons pour détection de droites avec Hough -->
   <widget class="QPushButton" name="DetectionDroite">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>240</y>
      <width>120</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Détection Droites</string>
    </property>
   </widget>

   <!-- Groupe de boutons pour la segmentation par couleur -->
   <widget class="QGroupBox" name="GroupeSegmentation">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <property name="title">
     <string>Segmentation Couleur</string>
    </property>
    <widget class="QCheckBox" name="Segmentation_Canal_R">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>150</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Rouge</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="Segmentation_Canal_V">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>150</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Vert</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="Segmentation_Canal_B">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>150</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Bleu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="SegmenterCouleur">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>150</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Segmenter</string>
     </property>
    </widget>
   </widget>

  </widget>

  <!-- Barre de menu et barre de statut -->
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
